contracts/trading/TradingModule.sol

```diff
@@ -229,6 +244,11 @@ contract TradingModule is Initializable, UUPSUpgradeable, ITradingModule {
         decimals = RATE_DECIMALS;
     }
 
+    /// @notice Check if the caller is allowed to execute the provided trade object
+    function canExecuteTrade(Trade calldata trade) external view override returns (bool) {
+        return tokenWhitelist[msg.sender][trade.sellToken].allowSell;
```

Comment:
Since this is called inside a delegatecall context shouldn't it be address(this)?

Reply:
I think using msg.sender here is ok. We are not doing a delegatecall on canExecuteTrade. We are calling it through PROXY.canExecuteTrade(trade), which makes msg.sender the vault address. I have some tests around this and they seem to be passing.